utils/src/server/rpc_module.rs

```diff
@@ -544,6 +545,41 @@ impl<Context: Send + Sync + 'static> RpcModule<Context> {
 		unsubscribe_method_name: &'static str,
 		callback: F,
 	) -> Result<(), Error>
+	where
+		Context: Send + Sync + 'static,
+		F: Fn(Params, SubscriptionSink, Arc<Context>) -> Result<(), Error> + Send + Sync + 'static,
+	{
+		self.register_subscription_inner(subscribe_method_name, None, unsubscribe_method_name, callback)
+	}
+
+	/// Similar to [`RpcModule::register_subscription`] but allows sending notifications to another method
+	/// than the actual subscription method name.
+	pub fn register_subscription_with_custom_notif<F>(
+		&mut self,
+		subscribe_method_name: &'static str,
+		custom_notif_method_name: &'static str,
+		unsubscribe_method_name: &'static str,
+		callback: F,
+	) -> Result<(), Error>
+	where
+		Context: Send + Sync + 'static,
+		F: Fn(Params, SubscriptionSink, Arc<Context>) -> Result<(), Error> + Send + Sync + 'static,
+	{
+		self.register_subscription_inner(
+			subscribe_method_name,
+			Some(custom_notif_method_name),
+			unsubscribe_method_name,
+			callback,
+		)
+	}
+
+	fn register_subscription_inner<F>(
+		&mut self,
+		subscribe_method_name: &'static str,
+		custom_notif_method_name: Option<&'static str>,
+		unsubscribe_method_name: &'static str,
+		callback: F,
+	) -> Result<(), Error>
```

Comment:
Any reason why this can't be the actual one-and-only public method? I'd even drop the `Option`, having duplicate `&'static str`s doesn't cost anything (or at very least doesn't cost more than having an extra `Option<&'static str>`, rustc will dedupe them and we save ourselves the extra branching in code).

```suggestion
	pub fn register_subscription<F>(
		&mut self,
		subscribe_method_name: &'static str,
		notification_method_name: &'static str,
		unsubscribe_method_name: &'static str,
		callback: F,
	) -> Result<(), Error>
```

Reply:
that should work but rather annoying API to use without the macros.

Reply:
If you think it's better to have separate methods I'd still drop the `Option` and merge it with `register_subscription_with_custom_notif`, and then alias `register_subscription` so that it calls `register_subscription_with_custom_notif` with a copy of the str, because that's also virtually free and you get rid of the branch.

Reply:
I did what you wished for let's see what @dvdplm says :)